crates/index-scheduler/src/versioning.rs

```diff
@@ -0,0 +1,68 @@
+use crate::{upgrade::upgrade_index_scheduler, Result};
+use meilisearch_types::{
+    heed::{types::Str, Database, Env, RoTxn, RwTxn},
+    milli::heed_codec::version::VersionCodec,
+    versioning,
+};
+
+/// The number of database used by queue itself
+const NUMBER_OF_DATABASES: u32 = 1;
+/// Database const names for the `IndexScheduler`.
+mod db_name {
+    pub const VERSION: &str = "version";
+}
+mod entry_name {
+    pub const MAIN: &str = "main";
+}
+
+#[derive(Clone)]
+pub struct Versioning {
+    pub version: Database<Str, VersionCodec>,
+}
+
+impl Versioning {
+    pub(crate) const fn nb_db() -> u32 {
+        NUMBER_OF_DATABASES
+    }
+
+    pub fn get_version(&self, rtxn: &RoTxn) -> Result<Option<(u32, u32, u32)>> {
+        Ok(self.version.get(rtxn, entry_name::MAIN)?)
+    }
+
+    pub fn set_version(&self, wtxn: &mut RwTxn, version: (u32, u32, u32)) -> Result<()> {
+        Ok(self.version.put(wtxn, entry_name::MAIN, &version)?)
+    }
+
+    pub fn set_current_version(&self, wtxn: &mut RwTxn) -> Result<()> {
+        let major = versioning::VERSION_MAJOR.parse().unwrap();
+        let minor = versioning::VERSION_MINOR.parse().unwrap();
+        let patch = versioning::VERSION_PATCH.parse().unwrap();
+        self.set_version(wtxn, (major, minor, patch))
+    }
+
+    /// Create an index scheduler and start its run loop.
+    pub(crate) fn new(env: &Env, db_version: (u32, u32, u32)) -> Result<Self> {
+        let mut wtxn = env.write_txn()?;
+        let version = env.create_database(&mut wtxn, Some(db_name::VERSION))?;
+        let this = Self { version };
+        let from = this.get_version(&wtxn)?.unwrap_or(db_version);
```

Comment:
Don't we need to write the index-scheduler version immediately if it is `None`. This is because at this point we already rewrote the version file?

Ideally we would:

1. read the index-scheduler version, defaulting to db version
2. write the db version as the index-scheduler version if it was None, committing the txn
3. Only then, write the VERSION file
4. We probably want to register the the task and change the index-scheduler version to latest in the same wtxn as well